spec/models/dossier_spec.rb

```diff
@@ -1345,29 +1345,40 @@
     let(:dossier_second_revision) { create(:dossier, procedure: procedure) }
 
-    before do
-      procedure.publish!
-      dossier
-      procedure.draft_revision.remove_type_de_champ(text_type_de_champ.stable_id)
-      procedure.draft_revision.add_type_de_champ(type_champ: TypeDeChamp.type_champs.fetch(:text), libelle: 'New text field')
-      procedure.draft_revision.find_or_clone_type_de_champ(yes_no_type_de_champ.stable_id).update(libelle: 'Updated yes/no')
-      procedure.update(published_revision: procedure.draft_revision, draft_revision: procedure.create_new_revision)
-      dossier.reload
-      procedure.reload
+    context "when procedure published" do
+      before do
+        procedure.publish!
+        dossier
+        procedure.draft_revision.remove_type_de_champ(text_type_de_champ.stable_id)
+        procedure.draft_revision.add_type_de_champ(type_champ: TypeDeChamp.type_champs.fetch(:text), libelle: 'New text field')
+        procedure.draft_revision.find_or_clone_type_de_champ(yes_no_type_de_champ.stable_id).update(libelle: 'Updated yes/no')
+        procedure.update(published_revision: procedure.draft_revision, draft_revision: procedure.create_new_revision)
+        dossier.reload
+        procedure.reload
+      end
+
+      it "should have champs from all revisions" do
+        expect(dossier.types_de_champ.map(&:libelle)).to eq([text_type_de_champ.libelle, datetime_type_de_champ.libelle, "Yes/no", explication_type_de_champ.libelle])
+        expect(dossier_second_revision.types_de_champ.map(&:libelle)).to eq([datetime_type_de_champ.libelle, "Updated yes/no", explication_type_de_champ.libelle, "New text field"])
+        expect(dossier.champs_for_export(dossier.procedure.types_de_champ_for_export).map { |(libelle)| libelle }).to eq([datetime_type_de_champ.libelle, "Updated yes/no", "New text field"])
+        expect(dossier.champs_for_export(dossier.procedure.types_de_champ_for_export)).to eq(dossier_second_revision.champs_for_export(dossier_second_revision.procedure.types_de_champ_for_export))
+      end
     end
 
-    it "should have champs from all revisions" do
-      expect(dossier.types_de_champ.map(&:libelle)).to eq([text_type_de_champ.libelle, datetime_type_de_champ.libelle, "Yes/no"])
-      expect(dossier_second_revision.types_de_champ.map(&:libelle)).to eq([datetime_type_de_champ.libelle, "Updated yes/no", "New text field"])
-      expect(dossier.champs_for_export(dossier.procedure.types_de_champ_for_export).map { |(libelle)| libelle }).to eq([datetime_type_de_champ.libelle, "Updated yes/no", "New text field"])
-      expect(dossier.champs_for_export(dossier.procedure.types_de_champ_for_export)).to eq(dossier_second_revision.champs_for_export(dossier_second_revision.procedure.types_de_champ_for_export))
+    context "when procedure brouillon" do
+      let(:procedure) { create(:procedure, :with_type_de_champ, :with_explication) }
+
+      it "should not contain explication type de champ" do
```

Comment:
Je pense qu'ici ça serait bien d'expliquer plus le "Pourquoi" que le "Quoi". Un truc du genre :

```suggestion
      it "should not contain non-exportable types de champ" do
```